### PR TITLE
[PoC] Integration tests with Apollo and PostgreSQL database

### PR DESCRIPTION
**Pull request by ggwadera**
This PR is a proof-of-concept for a possible way that we could do integration tests with Apollo Server and Prisma, connected with a live PostgreSQL database instance through Docker, as part of #670.

Included are 2 example tests files:
- `__tests__/integration/lessons.ts` - simple query for fetching the lessons and comparing to a snapshot
- `__tests__/integration/signup.ts` - more complex example that goes through the entire user creation process, verifying that the user was actually created on the database

References:
- https://www.apollographql.com/docs/apollo-server/testing/testing/
- https://www.prisma.io/docs/guides/testing/integration-testing#integration-testing

### CODE REVIEW

File: helpers/apolloClient.test.js [hunk 1]
```diff
@@ -17,7 +17,6 @@ jest.mock('@apollo/client', () => ({
     cache: { restore: jest.fn() }
   }))
 }))
-jest.mock('nodemailer')
```
Comment (hwong0305) [inline helpers/apolloClient.test.js:20]: I don't think we should remove mocking mail for unit tests. 

If we want to test the mail, we should only do it explicitly in the integration tests and e2e tests.

File: helpers/controllers/authController.test.js [hunk 2]
```diff
@@ -1,5 +1,4 @@
 jest.mock('bcrypt')
-jest.mock('nodemailer')
```
Comment (hwong0305) [inline helpers/controllers/authController.test.js:2]: same as above

File: helpers/controllers/passwordController.test.js [hunk 3]
```diff
@@ -4,7 +4,6 @@
 
 jest.mock('bcrypt')
 jest.mock('nanoid')
-jest.mock('nodemailer')
```
Comment (hwong0305) [inline helpers/controllers/passwordController.test.js:7]: same as above

File: helpers/mail.test.js [hunk 4]
```diff
@@ -6,12 +6,6 @@ import { getResetTemplate, getSignupTemplate } from './mailTemplate'
 import { sendResetEmail, sendSignupEmail } from './mail'
 import nodemailer from 'nodemailer'
 
-jest.mock('nodemailer', () => ({
-  createTransport: jest.fn().mockReturnValue({
-    sendMail: jest.fn().mockReturnValue((mailoptions, callback) => {})
-  })
-}))
-
```
Comment (hwong0305) [inline helpers/mail.test.js:14]: same as above

File: helpers/apolloClient.test.js (hunk 1 above)
Comment (ggwadera) [reply helpers/apolloClient.test.js:20]: I didn't remove nodemailer mocking, I made a manual mock under `__mocks__` that is automatically applied.

**Review by hwong0305 — APPROVED** [review]

**Comment by songz** [comment]
Closing because nobody has picked this up. Will be picked up with this issue: https://github.com/garageScript/c0d3-app/issues/670

**Pull request closed without merging**